### PR TITLE
Status API

### PR DESCRIPTION
**Pull request by mre**
This is a new endpoint, which returns the timestamps of the latest endpoint calls per account, e.g.

```json
{
    "account_id": 1,
    "latestUpdates": [
        "aggregate": "2021-01-01 00:00:00"
        "detailedStreams": "2021-01-01 00:00:00"
    ]
}
```

It is helpful for building an API status page in the future.

### CODE REVIEW

File: src/db/StatusRepository.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    constructor(pool: Pool) {
+        this.pool = pool
+    }
+
+    // Reads from the `updates` table and returns the latest status
+    // updates for the given accountId and endpoint.
+    // (Updates are stored in JSON format)
+    async getStatus(accountId: number, endpoint: string): Promise<Status> {
+        const query = `SELECT * FROM updates WHERE account_id = ? AND endpoint = ? ORDER BY updated_at DESC LIMIT 1`
```
Comment (woolfg) [inline src/db/StatusRepository.ts:20]: `ORDER BY timestamp` and you have to add a WHERE clause using the timestamp to be able to make use of the index and do not trigger a full table scan. I would recommend to specify timestamp > DATE_SUB(NOW(), INTERVAL 7 DAY) or something like this

Comment (mre) [reply src/db/StatusRepository.ts]: done

**Comment by mre** [comment]
> what about just showing the podcast_ids an the last action of it?

We would leak all podcast ids and the API could be a public endpoint, so I'm not sure if we can/should do this. 🤔

**Comment by mre** [comment]
Most likely this endpoint will be used by a status page for a single podcast, so providing the token shouldn't be a big deal.

**Comment by mre** [comment]
We agreed that we will use MySQL's event scheduler to remove events older than 7 days.

**Comment by mre** [comment]
Unfortunately the `event` keyword is not supported by PlanetScale.
See https://planetscale.com/docs/reference/mysql-compatibility and https://github.com/planetscale/discussion/discussions/265.

**Comment by woolfg** [comment]
can be testen locally using the `stack` repo?

**Comment by woolfg** [comment]
with the new changed in `stack` it should be possible to test it there. according to https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_event_scheduler it is ON by default anyways.

File: README.md [hunk 2]
```diff
@@ -39,6 +39,31 @@ It stores one unique vote for each episode per IP and user-agent pair.
...
+    "latestUpdates": [
+        "aggregate": "2021-01-01 00:00:00"
+        "detailedStreams": "2021-01-01 00:00:00"
+    ]
+}
+```
+
+To query it, use the following endpoint:
+
+`/status/:accountID`
```
Comment (woolfg) [inline README.md:58]: is already defined by the auth token, right?

File: db_schema/schema.sql [hunk 3]
```diff
@@ -212,4 +212,21 @@ CREATE TABLE IF NOT EXISTS feedbackComment (
...
+  endpoint VARCHAR(64) NOT NULL,
+  timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  update_data JSON NOT NULL,
+  PRIMARY KEY (timestamp, account_id, endpoint)
+);
+
+-- automatically delete all updates older than 7 days
+CREATE EVENT IF NOT EXISTS updates_cleanup
+ON SCHEDULE EVERY 1 DAY
+DO DELETE FROM updates WHERE timestamp < (CURRENT_TIMESTAMP - INTERVAL 7 DAY);
```
Comment (woolfg) [inline db_schema/schema.sql:232]: `DATE_SUB(NOW(), INTERVAL 7 DAY)`

File: db_schema/schema.sql [hunk 4]
```diff
@@ -212,4 +212,21 @@ CREATE TABLE IF NOT EXISTS feedbackComment (
...
+);
+
+-- store latest update events
+-- timestamp is used to identify the update
+-- contains JSON with the update data
+DROP TABLE IF EXISTS updates;
+CREATE TABLE updates (
+  account_id INTEGER NOT NULL,
+  endpoint VARCHAR(64) NOT NULL,
+  timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
Comment (woolfg) [inline db_schema/schema.sql:224]: wouldn't recommend to use keywords as field names (timestamp TIMESTAMP), maybe just `created`?

**Review by woolfg — APPROVED** [review]
looks good. added some minor remarks. in general it would be nicer to be able to check the status without auth request and just see a table of account_ids, but would merge first and improce later.

File: README.md (hunk 2 above)
Comment (mre) [reply README.md:58]: True. Fixed.

File: db_schema/schema.sql (hunk 3 above)
Comment (mre) [reply db_schema/schema.sql:232]: nice

**Comment by mre** [comment]
* Removed authentication
* Changed response to reply with updates of all accounts.
* Renamed `timestamp` field to `created`
* Updated the field name in PlanetScale (dev only).
* Tested locally

Example response:

```json
{
  "1": {
    "latestUpdates": {
      "metadata": "2022-12-23T15:03:47.000Z",
      "detailedStreams": "2022-12-23T15:05:14.000Z",
      "listeners": "2022-12-23T15:05:17.000Z",
      "aggregate": "2022-12-23T15:05:12.000Z",
      "followers": "2022-12-23T15:04:00.000Z",
      "episodes": "2022-12-23T15:04:03.000Z",
      "performance": "2022-12-23T15:05:19.000Z"
    }
  },
  "2": {
    "latestUpdates": {
      "listeners": "2023-01-30T11:30:38.000Z"
    }
  }
}
```

**Review by woolfg — APPROVED** [review]

**Pull request merged**